mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp

```diff
@@ -2695,7 +2620,7 @@ static Value buildDivOp(OpBuilder &builder, Location loc, Value numerator,
 /// 4. Divide z and l. This gives the N-dimensional softmax.
 ///    softmax = z / l
 ///
-FailureOr<SmallVector<Value>> SoftmaxOp::decomposeOperation(OpBuilder &b) {
+FailureOr<DecompositionResult> SoftmaxOp::decomposeOperation(OpBuilder &b) {
```

Comment:
Current logic assumes softmax is at tensor-level and crashes on memref.
Ideally, it should work on both abstractions but for now it could just fail gracefully.

Either way, I'd add a test case on memrefs too.

Reply:
Yup, will do.